### ISSUE TITLE
Compile Error: no member named 'spherical_mercator' in namespace 'mapnik::vector_tile_impl'

### ISSUE THREAD

**Issue by tonny2v** [mapnik/node-mapnik#851]
On Mac OS, I succeeded in building all dependencies but when I finalize the compiling. I got the following errors:
```bash
../src/mapnik_vector_tile.cpp:5124:53: error: expected ';' after expression
        mapnik::vector_tile_impl::spherical_mercator merc(closure->d->tile_size());
                                                    ^
                                                    ;
../src/mapnik_vector_tile.cpp:5124:35: error: no member named 'spherical_mercator' in namespace 'mapnik::vector_tile_impl'
        mapnik::vector_tile_impl::spherical_mercator merc(closure->d->tile_size());
        ~~~~~~~~~~~~~~~~~~~~~~~~~~^
../src/mapnik_vector_tile.cpp:5124:54: error: use of undeclared identifier 'merc'
        mapnik::vector_tile_impl::spherical_mercator merc(closure->d->tile_size());
                                                     ^
../src/mapnik_vector_tile.cpp:5128:13: error: use of undeclared identifier 'merc'
            merc.xyz(closure->x,closure->y,closure->z,minx,miny,maxx,maxy);
            ^
../src/mapnik_vector_tile.cpp:5132:13: error: use of undeclared identifier 'merc'
            merc.xyz(closure->d->get_tile()->x(),
            ^
1 warning and 5 errors generated.
make[2]: *** [Release/obj.target/mapnik/src/mapnik_vector_tile.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/opt/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)
gyp ERR! stack     at ChildProcess.emit (events.js:160:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:209:12)
gyp ERR! System Darwin 17.4.0
gyp ERR! command "/opt/local/bin/node" "/opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "build" "--ENABLE_GLIBC_WORKAROUND=true" "--loglevel=error" "--clang" "--module=/Volumes/SSD/github/node-mapnik/lib/binding/mapnik.node" "--module_name=mapnik" "--module_path=/Volumes/SSD/github/node-mapnik/lib/binding"
gyp ERR! cwd /Volumes/SSD/github/node-mapnik
gyp ERR! node -v v9.5.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute '/opt/local/bin/node /opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --ENABLE_GLIBC_WORKAROUND=true --loglevel=error --clang --module=/Volumes/SSD/github/node-mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/Volumes/SSD/github/node-mapnik/lib/binding' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/Volumes/SSD/github/node-mapnik/node_modules/node-pre-gyp/lib/util/compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:160:13)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:943:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:220:5)
node-pre-gyp ERR! System Darwin 17.4.0
node-pre-gyp ERR! command "/opt/local/bin/node" "/Volumes/SSD/github/node-mapnik/node_modules/.bin/node-pre-gyp" "configure" "build" "--ENABLE_GLIBC_WORKAROUND=true" "--loglevel=error" "--clang"
node-pre-gyp ERR! cwd /Volumes/SSD/github/node-mapnik
node-pre-gyp ERR! node -v v9.5.0
node-pre-gyp ERR! node-pre-gyp -v v0.6.39
node-pre-gyp ERR! not ok
Failed to execute '/opt/local/bin/node /opt/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js build --ENABLE_GLIBC_WORKAROUND=true --loglevel=error --clang --module=/Volumes/SSD/github/node-mapnik/lib/binding/mapnik.node --module_name=mapnik --module_path=/Volumes/SSD/github/node-mapnik/lib/binding' (1)
make[1]: *** [release_base] Error 1
make: *** [release] Error 2

**Comment by flippmoke** [comment]
It looks like you are using the latest version of `mapnik-vector-tile` which is `v1.6.0`, or you might be using the `master` branch on each. We do have prebuilt binaries for common linux uses available via `npm install`. Do they not work properly for you?

Additionally, the state of dependencies is a little wonky right now (I am sorry). 

Master `mapnik`, master `node-mapnik`, and `mapnik-vector-tile` are under current development right now and no release of all them together has been achieved. We are waiting on an official mapnik release to begin to tag and release node-mapnik and mapnik-vector-tile.

If you want to use the last released Mapnik (v3.0.x series), there are `v3.0.x` branches in all of the stated libraries. The `v3.0.x` series should come to an end with a new official release of master on `mapnik`. 

Also note that I recently updated `mapnik-vector-tile`, but I am still working towards patching in correctly the changes into `node-mapnik` correctly. See https://github.com/mapnik/node-mapnik/pull/850

Probably the easiest thing you can do if you need to custom compile `node-mapnik` is to check out an older tag and build it. By default it should pull in the required binaries if you simply run `make`

**Comment by tonny2v** [comment]
Thanks very much. I will try an older version.
